### DISCUSSION TITLE
Workflow for use Flowershow as wiki on GitHub?

### DISCUSSION THREAD

**Discussion by David7ce** [flowershow/flowershow#569] [General and Help] [answered]
First thanks to the devs, @olayway and @rufuspollock .

Since I've discovered Obsidian.md and the publish plugin I've tried to achieve the best workflow to create a free wiki, based on an Obsidian Vault that can be deployed easily as a *static website* without backend. So people can edit files in markdown and download only the content.

I've looked for a lot of Obsidian Publish alternatives here's a list https://raindrop.io/Interneto/obsidian-plubish-generator-29969898

I also have tried with SSG like Obsidian Digital Garden (based on 11Ty), some HUGO themes, VitePress, Retype, Docsify.

I am using Obsidian DG but the problem is that is not made to build in local with output folder, it was thinked to push directly to a cloud server like Vercel via an API.
Retype is also very easy to use, just create the static website based on the folder structure but is limited to 100 files or pay annually and it doesn't generate an output folder.

My goal is to use a Static Site Generator that converts an Obsidian vault into a wiki that can be editable with the left sidebar with directory structure, the right sidebar content, edit button, markdown compatibility, etc.

Flowershow maybe could satisfied my necessities.

I only have one doubt: Can I push everything from Flowershow to GitHub instead of Vercel? And do I need and 'index.md' for every directory?

BTW node_modules is 1GiB of data if you build locally, wow...

**Comment by rufuspollock** [comment, marked as answer]
@David7ce thanks for writing up the idea and it sounds cool.

> I only have one doubt: Can I push everything from Flowershow to GitHub instead of Vercel? And do I need and 'index.md' for every directory?

yes you can push everything to github rather than Vercel. we use it like that ourselves in various places. note however it won't render nicely purely on github (though you can build through github pages rather than e.g. vercel).

> BTW node_modules is 1GiB of data if you build locally, wow...

Yes, i know ... One option is to look around for the big culprits e.g. do you need mermaid? if not you can cut that from the `package.json`
